core/src/commonMain/kotlin/com/ustadmobile/core/controller/ClazzAssignmentEditPresenter.kt

```diff
@@ -136,20 +266,32 @@ class ClazzAssignmentEditPresenter(context: Any,
                 return@launch
             }
 
+
             if (entity.caLateSubmissionType == ClazzAssignment.ASSIGNMENT_LATE_SUBMISSION_ACCEPT ||
                     entity.caLateSubmissionType == ClazzAssignment.ASSIGNMENT_LATE_SUBMISSION_PENALTY) {
 
-                if (entity.caDeadlineDate == Long.MAX_VALUE) {
+                if (entity.caDeadlineDate == Long.MAX_VALUE || view.deadlineTime == 0L) {
```

Comment:
Let's show all errors for the user as soon as they click submit (e.g. so they don't fix one thing. click again, then see the next error)